src/charm.py

```diff
@@ -116,14 +138,33 @@ def _mysql_layer(self, service_name: str) -> Layer:
     # ---- Properties
     @property
     def _config(self):
-        """Configuration data for MySQL authentication."""
+        """Configuration data for MySQL."""
         config = {
+            "innodb_cluster_name": self.model.config["innodb_cluster_name"],
             "mysql_user": "mysql_user",
-            "mysql_password": self._generate_password(),
-            "mysql_root_password": self._generate_password(),
+            "mysql_password": self._mysql_password,
+            "mysql_root_password": self._root_password,
```

Comment:
As above. Get this from self.model.config['mysql_password'] and self.model.config['root_password']. This is a much more reliable source of truth than the StoredState.